google/cloud/spanner_v1/_helpers.py

```diff
@@ -47,6 +48,45 @@ def _try_to_coerce_bytes(bytestring):
         )
 
 
+def _merge_query_options(base, merge):
+    """Merge higher precedence QueryOptions with current QueryOptions.
+
+    :type base:
+        :class:`google.cloud.spanner_v1.proto.ExecuteSqlRequest.QueryOptions`
+        or :class:`dict` or None
+    :param base: The current QueryOptions that is intended for use.
+
+    :type merge:
+        :class:`google.cloud.spanner_v1.proto.ExecuteSqlRequest.QueryOptions`
+        or :class:`dict` or None
+    :param merge:
+        The QueryOptions that have a higher priority than base. These options
+        should overwrite the fields in base.
+
+    :rtype:
+        :class:`google.cloud.spanner_v1.proto.ExecuteSqlRequest.QueryOptions`
+        or None
+    :returns:
+        QueryOptions object formed by merging the two given QueryOptions.
+        If the resultant object only has empty fields, returns None.
+    """
+    combined = base or ExecuteSqlRequest.QueryOptions()
```

Comment:
nits: probably think a better way to merge in a more generic way (less code if we add new fields). Currently, we need to keep add new fields. 

E.g., convert to `dict` first, merge, and then create `ExecuteSqlRequest.QueryOptions`.

Reply:
I discovered the MergeFrom protobuf function so I refactored to use that.